packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -109,6 +106,24 @@ abstract class ResidentRunner {
     await currentView.uiIsolate.flutterToggleDebugPaintSizeEnabled();
   }
 
+  Future<Null> _screenshot() async {
+    File outputFile = getUniqueFile(fs.currentDirectory, 'flutter', 'png');
+    try {
+      if (vmService != null)
+        await vmService.vm.refreshViews();
+      if (isRunningDebug)
+        await currentView.uiIsolate.flutterDebugAllowBanner(false);
```

Comment:
The problem is that screenshot.dart has a very different context. It doesn't have a VMService connection for example. We could provide a way to specify the observatory port, etc, but that seems more work than this is worth.